### PR TITLE
Issue 5546, 5547:  Fix flaky tests

### LINKED ISSUE #5546: Flaky test: DynamicRestApiTest
**Problem description**
SystemTest failure : DynamicRestApiTest

```
2021-01-18 20:45:22,784 2531 [Time-limited test] ERROR i.p.t.system.SingleJUnitTestRunner - Test listScopes failed with exception. 
java.lang.NullPointerException: password is marked non-null but is null  
at io.pravega.shared.security.auth.DefaultCredentials.<init>(DefaultCredentials.java:33)                                                                                                               at io.pravega.test.system.DynamicRestApiTest.listScopes(DynamicRestApiTest.java:120)                                                                                                                   at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                                                                                                      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)                                                                                                                      at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)                                                                                                                  at java.base/java.lang.reflect.Method.invoke(Unknown Source)      
```

**Problem location**
DynamicRestApiTest

**Suggestions for an improvement**
Fix it.

### LINKED ISSUE #5547: Sporadic failure in EndToEndTruncationTest
**Problem description**
```
io.pravega.test.integration.endtoendtest.EndToEndTruncationTest > testDeleteStreamWhileReading FAILED
    java.lang.AssertionError: expected:<50> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at io.pravega.test.integration.endtoendtest.EndToEndTruncationTest.testDeleteStreamWhileReading(EndToEndTruncationTest.java:676)
```

**Problem location**
EndToEndTruncationTest 

**Suggestions for an improvement**
Fix the problem so that it is not flakey

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  

- Issue 5546 Ensure system test DynmamicRestApi works with auth enabled.
- Issue 5008: Remove incorrect fix with SO_REUSEADDR .
- Issue 5547: Solve flaky test EndToEndTruncationTest > testDeleteStreamWhileReading

**Purpose of the change**  
Fixes #5546 
Fixes #5547 
Remove incorrect fix for #5008 

**What the code does**  
- Issue 5546 Ensure system test for DynamicRestApi works with auth enabled.
- Issue 5008: Remove incorrect fix with SO_REUSEADDR .
The previous fix was based on pattern used in the camel project and previous fix would not fix the sporadic BindException observed in our test runs.
Also, the correct usage pattern  to set reuse address is 

``` 
                ServerSocket serverSocket = new ServerSocket();
                //Enabling SO_REUSEADDR prior to binding the socket using bind(SocketAddress) allows the socket
                //to be bound even though a previous connection is in a timeout state.
                serverSocket.setReuseAddress(true);
                serverSocket.bind(new InetSocketAddress(candidatePort));
                serverSocket.close();
```
- Issue 5547: Solve flaky test EndToEndTruncationTest > testDeleteStreamWhileReading

**How to verify it**  
All the tests should continue to pass.

### CODE REVIEW

**Comment by shrids** [comment]
System test runs are successful 
With auth enabled: job/test-pravega-continuous-1/1182/ 
Without auth enabled: job/test-pravega-continuous/3456/

File: test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndTruncationTest.java [hunk 1]
```diff
@@ -673,7 +674,7 @@ public void testDeleteStreamWhileReading() {
         groupManager.createReaderGroup(readerGroup, ReaderGroupConfig.builder().automaticCheckpointIntervalMillis(500).stream(Stream.of(scope, streamName)).build());
         @Cleanup
         EventStreamReader<String> reader = clientFactory.createReader(String.valueOf(0), readerGroup, new UTF8StringSerializer(), ReaderConfig.builder().build());
-        assertEquals(totalEvents / 2, ReadWriteUtils.readEvents(reader, totalEvents / 2, 0));
+        assertEquals(totalEvents / 2, ReadWriteUtils.readEventsUntil(reader, eventRead -> true, totalEvents / 2, 0));
```
Comment (shiveshr) [inline test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndTruncationTest.java:677]: sorry, i dont understand the implication here .. when predicate is always true, then we are effectively using the the original method itself.. so what is being fixed here?

**Review by shiveshr — APPROVED** [review]
LGTM, except one question

File: test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndTruncationTest.java (hunk 1 above)
Comment (shrids) [reply test/integration/src/test/java/io/pravega/test/integration/endtoendtest/EndToEndTruncationTest.java:677]: the previous method stopped reading if a null returned by the reader, which can be due to any reason including a slower response on the test infra. Now the test tries until all the expected events are read or a test timeout is observed.

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**